extensions/replication/tasks/ReplicateObject.js

```diff
@@ -132,29 +133,101 @@ class ReplicateObject extends BackbeatTask {
         }, cb);
     }
 
-    _publishReplicationStatus(updatedSourceEntry, params, cb) {
+    _getMetadata(entry, log, cb) {
+        this.retry({
+            actionDesc: 'get metadata from source',
+            logFields: { entry: entry.getLogInfo() },
+            actionFunc: done => this._getMetadataOnce(entry, log, done),
+            shouldRetryFunc: err => err.retryable,
+            log,
+        }, cb);
+    }
+
+    _getMetadataOnce(entry, log, cb) {
+        log.debug('getting metadata', {
+            where: 'source',
+            entry: entry.getLogInfo(),
+            method: 'ReplicateObject._getMetadataOnce',
+        });
+        const cbOnce = jsutil.once(cb);
+
+        const req = this.backbeatDest.getMetadata({
```

Comment:
I think this is meant to be `this.backbeatSource`?

Reply:
Ah right, good catch!